client/src/templates/Challenges/classic/Show.js

```diff
@@ -249,6 +256,13 @@ class ShowClassic extends Component {
     );
   }
 
+  renderSpaceForFlashMessage = () => {
+    const { flashMessage } = this.props;
+    const hasFlashMessage = !!flashMessage.message;
+    const flashMessageHeight = '56px';
```

Comment:
```suggestion
    const flashMessageHeight = '56px';
```
This should also be subtracted from the height of the reflex container, so that it doesn't get an extra scrollbar.

Reply:
Just updated this.  The new code is setting the Flash message height as a css variable which is used to subtract the height from the flex container and add space at the top of the browser for every flash message.  I moved the code out of Show.js and it is all being handled in Flash/index.js.  The height of the desktop and mobile reflex containers were already being calculated using the "--header-height" css var so I just added the flex height to the equation.  As far as I could find,  that is the best solution.

Reply:
The more I think about this, the more I feel like this would be best as a separate PR.  Reason being, the flash message functionality is independent of code storage (it just happens that we're using flash messages to tell people what's happened to their code). @ahmadabdolsaheb @raisedadead what do you reckon? @jakethecoder95 please don't make any changes until someone else has commented.

As for the approach, I think it's interesting, but I don't think it will work for small screens (where the message wraps and the height isn't always 56px).  We need to come up with a responsive solution.

Reply:
Sounds good.  I'll wait to do any updates till that is discussed.  

And yeah that was an issue with on mobile but I think I solved it in the Flash/index.js file updates.  The css variable height is set based on the height of the flash message, "document.querySelector('.flash-message').offsetHeight" line 14

Reply:
> And yeah that was an issue on mobile but I think I solved it in the Flash/index.js file updates. The css variable height is set based on the height of the flash message, "document.querySelector('.flash-message').offsetHeight" line 14

Ah, sorry, I didn't look closely enough at the code.  I was just playing around and noticed it occasionally would cover the tabs.  A bit more investigating shows that it's only a problem when you resize or when there's already a different size flash message visible.  For example:
![Flash](https://user-images.githubusercontent.com/15801806/70826986-8e957880-1de8-11ea-9019-4e1c7a4b82f3.gif)

which is not going to be an issue most of the time.

Reply:
Oh, so as of now mobile users shouldn't even see the flash message.  In code-storage-epic.js it only creates a flash message when the user manually saves with Ctrl/Cmd + S.  I did this because it is a confusing UX when the code submits successfully and the flash message is still on the screen when the user enters the new challenge.  We could easily fix that though by removing any flash messages when the code is being deleted from localStorage.  That will require more changes though so I'll wait to see if this should be in a separate PR